tests/zfs-tests/tests/functional/events/zed_fd_spill.ksh

```diff
@@ -37,53 +37,32 @@ verify_runnable "both"
 function cleanup
 {
 	log_must rm -rf "$logdir"
+	log_must rm "/tmp/zts-zed_fd_spill-logdir"
 	log_must zed_stop
 }
 
 log_assert "Verify ZEDLETs inherit only the fds specified"
 log_onexit cleanup
 
-logdir="$(mktemp -d)"
-log_must cc -xc -o${ZEDLET_DIR}/all-dumpfds << EOF
-#include <sys/types.h>
-#include <sys/wait.h>
-#include <fcntl.h>
-#include <unistd.h>
-#include <stdlib.h>
-#include <stdio.h>
+logdir="$(mktemp -d)/"
+printf "%s" "$logdir" > /tmp/zts-zed_fd_spill-logdir
 
-int main() {
-	if(fork()) {
-		int err;
-		wait(&err);
-		return err;
-	}
-
-	char buf[4096];
-	snprintf(buf, sizeof (buf), "$logdir/%d", getppid());
-	dup2(creat(buf, 0644), STDOUT_FILENO);
-
-	// Fault injexion!
-	snprintf(buf, sizeof (buf), "%d", (getppid() % 20) + 4);
-	puts(buf);
-
-	snprintf(buf, sizeof (buf), "/proc/%d/fd", getppid());
-	execlp("ls", "ls", buf, NULL);
-	_exit(127);
-}
-EOF
+self="$(readlink -f "$0")"
+log_must ln -s "${self%/*}" "${ZEDLET_DIR}/all-dumpfds"
 
 log_must zpool events -c
 log_must zed_stop
 log_must zed_start
 
 truncate -s 0 $ZED_DEBUG_LOG
-log_must zpool scrub $testpool
+log_must zpool scrub $TESTPOOL
 log_must zfs set compression=off $TESTPOOL/$TESTFS
 wait_scrubbed $TESTPOOL
```

Comment:
I'd suggest adding a `log_must` before the `truncate` above and the `wait_scrubbed` here.  This will ensure when the command/function executes we'll get a line in the log for it.

Reply:
Sure, will include in the next iteration, I'd just modelled this after `zed_rc_filter` which doesn't use it for them.